### ISSUE TITLE
Accept pre-trained model as input

### ISSUE THREAD

**Issue by borauyar** [BIMSBbioinfo/flexynesis#105]
In the current setup, both train/test datasets must be available at startup, which is necessary to make sure both train/test datasets contain the same set of features and clinical labels. However, we should be able to have a pre-trained dataset and a new test dataset so that the model is evaluated on the test dataset. 

Challenges:
- Not all features or data modalities that were available during training will be available on the new/unseen dataset. Features that don't exist in the train data could be just removed, but features that don't exist in the test data may need a different approach (maybe the model could tolerate small portion of missing features, but the evaluation should be rejected if too many features are missing)
- The trained model should be shipped together with the data importer which contains the required data processing functions (e.g. scaling factors learned from the training data, or encoders for the metadata labels available for the training data)

**Comment by amitpande74** [comment]
Implemented the inference CLI per acceptance criteria in PR #109.

What’s in:

Accept pretrained model path + artifacts (.joblib) + test dataset folder.

Loads InferenceArtifacts, logs a summary, and writes a receipt to outdir/prefix.inference_receipt.json so CI/users can validate wiring end-to-end.

Uses keyword-only run_inference(model, artifacts_path, data_path_test, outdir, prefix) contract (model passed as a lightweight dict with checkpoint path + device).

Usage

python -m flexynesis.main \
  --pretrained_model /path/model.pth \
  --artifacts /path/preproc.joblib \
  --data_path_test /path/testdata \
  --prefix smoke


Output: results/smoke.inference_receipt.json

Notes

--join_key is optional for now (warns if missing).

Feature harmonization + predictions will follow in this branch.

Summary

Adds inference-only CLI in flexynesis/main.py.

Accepts pretrained model path + artifacts (.joblib) + test dataset and calls run_inference (keyword-only).

Writes results/<prefix>.inference_receipt.json to confirm parameters and artifacts wiring.

Contract
run_inference(model: nn.Module|dict, artifacts_path: str, data_path_test: str, outdir: str, prefix: str) -> None

Usage

python -m flexynesis.main \
  --pretrained_model /path/model.pth \
  --artifacts /path/preproc.joblib \
  --data_path_test /path/testdata \
  --prefix smoke


Notes

--join_key optional for now (warning only).

This PR focuses on the CLI + plumbing; feature alignment + actual predictions will come next.

Closes #105.
